solc/CommandLineParser.cpp

```
@@ -790,7 +796,16 @@ General Information)").c_str(),
 		m_options.output.dir = m_args.at(g_strOutputDir).as<string>();
 
 	m_options.output.overwriteFiles = (m_args.count(g_strOverwrite) > 0);
-	m_options.formatting.prettyJson = (m_args.count(g_strPrettyJson) > 0);
+
+	if (m_args.count(g_strPrettyJson) > 0)
+	{
+		m_options.formatting.prettyJson.format = JsonPrintingFormat::Format::Pretty;
+		if (m_args.count(g_strPrettyJsonIndent))
+		{
+			m_options.formatting.prettyJson.indent = m_args[g_strPrettyJsonIndent].as<uint32_t>();
+		}
+	}
```

I think we should report an error if `--pretty-json-indent` is used without `--pretty-json`. Alternatively, `--pretty-json-indent` could automatically enable `--pretty-json`.